src/server/scripts/BrokenIsles/NeltharionsLair/boss_rokmora.cpp

```diff
@@ -0,0 +1,569 @@
+        if (!navarrogg)
+            return;
+
+        conversation->AddActor(CONVO_ACTOR_NAVARROGG, CONVO_ACTOR_ID_NAVARROGG, navarrogg->GetGUID());
+    }
+
+    void OnStart() override
+    {
+        if (Milliseconds const* ularoggMoveStartTime = conversation->GetLineStartTime(DEFAULT_LOCALE, CONVO_ULAROGG_INTRO_LINE_03))
+            _events.ScheduleEvent(EVENT_ROKMORA_INTRO_MOVE, *ularoggMoveStartTime + 3s);
+    }
+
+    void OnUpdate(uint32 diff) override
+    {
+        _events.Update(diff);
+
+        switch (_events.ExecuteEvent())
+        {
+            case EVENT_ROKMORA_INTRO_MOVE:
+            {
+                Creature* ularogg = conversation->GetActorCreature(CONVO_ACTOR_ID_ULAROGG);
+                Creature* navarrogg = conversation->GetActorCreature(CONVO_ACTOR_ID_NAVARROGG);
+                if (!ularogg || !navarrogg)
+                    break;
+
+                ularogg->GetMotionMaster()->MovePath(PATH_ULAROGG, false);
+                navarrogg->GetMotionMaster()->MovePath(PATH_NAVARROGG, false);
+                break;
+            }
+            default:
+                break;
+        }
+    }
+
+private:
+    EventMap _events;
+};
+
+// 209394 - Boss 01 Intro Emerge
+class spell_rokmora_emerge : public SpellScript
+{
+    bool Validate(SpellInfo const* /*spellInfo*/) override
+    {
+        return ValidateSpellInfo({ SPELL_UPDATE_INTERACTIONS });
+    }
+
+    void HandleShatter() const
+    {
+        Creature* creatureCaster = GetCaster()->ToCreature();
+        InstanceScript* instance = creatureCaster->GetInstanceScript();
+        if (!instance)
+            return;
+
+        creatureCaster->CastSpell(creatureCaster, SPELL_UPDATE_INTERACTIONS, TRIGGERED_IGNORE_CAST_IN_PROGRESS | TRIGGERED_DONT_REPORT_CAST_ERROR);
+
+        instance->SetData(DATA_ROKMORA_INTRO_STATE, DONE);
+    }
+
+    void Register() override
+    {
+        AfterCast += SpellCastFn(spell_rokmora_emerge::HandleShatter);
+    }
+};
+
+class spell_rokmora_emerge_aura : public AuraScript
+{
+    void OnApply(AuraEffect const* /*aurEff*/, AuraEffectHandleModes /*mode*/) const
+    {
+        GetCaster()->ToCreature()->AddUnitState(UNIT_STATE_ROOT);
+    }
+
+    void OnRemove(AuraEffect const* /*aurEff*/, AuraEffectHandleModes /*mode*/) const
+    {
+        GetCaster()->ToCreature()->ClearUnitState(UNIT_STATE_ROOT);
+    }
+
+    void Register() override
+    {
+        OnEffectApply += AuraEffectApplyFn(spell_rokmora_emerge_aura::OnApply, EFFECT_0, SPELL_AURA_MOD_STUN, AURA_EFFECT_HANDLE_REAL);
+        OnEffectRemove += AuraEffectRemoveFn(spell_rokmora_emerge_aura::OnRemove, EFFECT_0, SPELL_AURA_MOD_STUN, AURA_EFFECT_HANDLE_REAL);
+    }
+};
+
+// 134730 - Update Interactions
+class spell_rokmora_update_interactions : public SpellScript
+{
+    void HandleScript(SpellEffIndex /*effIndex*/) const
+    {
+        Creature* creatureCaster = GetCaster()->ToCreature();
+        creatureCaster->SetImmuneToPC(false);
```

Comment:
SetImmuneToPC is also part of unit there's no need to cast it to creature if you're not checking against it

Reply:
it can crash too if a player casts this